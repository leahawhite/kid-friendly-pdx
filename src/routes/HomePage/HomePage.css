.hero {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 95%;
  margin: 0 auto;
  height: 79vh;
  background-color: #b3f0ff;
}

p {
  font-size: 18px;
  font-family: 'Source Sans Pro';
}

h1 {
  font-size: 32px;
  padding: 0;
  margin: 0;
  font-family: 'Fredoka One';
}

.places-list {
  display: flex;
  flex-direction: column;
  align-content: center;
}

@media only screen and (min-width: 500px) {
  h1 {
    font-size: 40px;
  }
}