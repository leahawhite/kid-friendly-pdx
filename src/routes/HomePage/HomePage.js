import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import SearchBar from '../../components/SearchBar/SearchBar';
import './HomePage.css';

export default class HomePage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      searchTerm: '',
      category: '',
      neighborhood: '',
      fireRedirect: false,
      error: null,
    }
  }

  handleUpdateSearch = event => {
    this.setState({
      searchTerm: event.target.value
    })
  }

  handleUpdateCategory = event => {
    this.setState({
      category: event.target.value
    })
  }

  handleUpdateNeighborhood = event => {
    this.setState({
      neighborhood: event.target.value
    })
  }

  handleSubmit = event => {
    event.preventDefault();
    this.setState({ fireRedirect: true })
  }
  
  render() {
    const { searchTerm, category, neighborhood, fireRedirect } = this.state
    return (
        <div className="hero">
          <header role="banner">
            <h1><Link to='/'>Kid-Friendly PDX</Link></h1>
            <p>Find the most welcoming spots in town.</p>
          </header>
          <SearchBar 
            searchTerm={searchTerm}
            category={category}
            neighborhood={neighborhood}
            fireRedirect={fireRedirect}
            handleUpdateSearch={this.handleUpdateSearch}
            handleUpdateCategory={this.handleUpdateCategory}
            handleUpdateNeighborhood={this.handleUpdateNeighborhood}
            handleSubmit={this.handleSubmit}
          />
          {fireRedirect && 
            <Redirect to={{
              pathname: '/places',
              state: { searchTerm, category, neighborhood } 
            }}
          />}
        </div>  
    )
  }
}