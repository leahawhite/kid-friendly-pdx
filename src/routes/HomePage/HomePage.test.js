import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomePage from './HomePage';

describe('HomePage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<HomePage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})