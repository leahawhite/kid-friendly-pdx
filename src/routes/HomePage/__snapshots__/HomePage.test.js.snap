// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders UI as expected 1`] = `
<div
  className="hero"
>
  <header
    role="banner"
  >
    <h1>
      <Link
        to="/"
      >
        Kid-Friendly PDX
      </Link>
    </h1>
    <p>
      Find the most welcoming spots in town.
    </p>
  </header>
  <SearchBar
    category=""
    fireRedirect={false}
    handleSubmit={[Function]}
    handleUpdateCategory={[Function]}
    handleUpdateNeighborhood={[Function]}
    handleUpdateSearch={[Function]}
    neighborhood=""
    searchTerm=""
  />
</div>
`;
